Source/PINDiskCache.m

```diff
@@ -1116,6 +1257,26 @@ - (void)trimToSizeByDate:(NSUInteger)trimByteCount
     [self trimDiskToSizeByDate:trimByteCount];
 }
 
+- (void)removeExpiredObjects
+{
+    [self lockForWriting];
+        NSDate *now = [NSDate date];
+        NSMutableArray<NSString *> *expiredObjectKeys = [NSMutableArray array];
+        [_metadata enumerateKeysAndObjectsUsingBlock:^(NSString * _Nonnull key, PINDiskCacheMetadata * _Nonnull obj, BOOL * _Nonnull stop) {
+            NSTimeInterval ageLimit = obj.ageLimit > 0.0 ? obj.ageLimit : self->_ageLimit;
+            NSDate *expirationDate = [obj.date dateByAddingTimeInterval:ageLimit];
```

Comment:
Ok, so I think there's a problem: now we're always setting date. It should probably be renamed lastModifiedDate and a new property createdDate should be added? Because objects shouldn't have their expiration lengthened when they're accessed, but they should live longer, in relation to other objects, than other objects which haven't expired but were accessed later? This behavior will have to be added / updated in PINMemoryCache as well?

Reply:
Good call. I modified the code using your suggestions. PTAL when you get a chance.